### ISSUE TITLE
ExpandMode for only certain levels of the tree

### ISSUE THREAD

**Issue by Nokhoneryu** [Crazyht/ngx-tree-select#165]
I have a very specific case on my app, in which i need to have only the first level of the hierarchy expanded and all the other collapsed. Is it possible?

**Comment by Crazyht** [comment]
Hi @Nokhoneryu,

now it's not possible and I don't have time to do this (frankly I don't have time to spend for OSS project since few month).

**Comment by Nokhoneryu** [comment]
Of course, but can you give me some hints on where i can tweak this to my project. I can then fork it so you can merge on the library if you want

**Comment by Crazyht** [comment]
I think you need : 
- Add @input Parameter (i.e. ExpandLevel: number)
- Add By Level Mode 
- add level parameter to setExpandForList (select.service.ts) with default value to 0
then you can handle you new mode depending of current recursion level

**Comment by Nokhoneryu** [comment]
Alright! I'll look into it! Many Thanks!
